powerauth-webflow/src/main/java/io/getlime/security/powerauth/app/webflow/configuration/SecurityConfiguration.java

```diff
@@ -19,67 +19,131 @@
+
+    @Autowired
+    public SecurityConfiguration(SecurityContextRepository securityContextRepository, AuthenticationManagementService authenticationManagementService) {
+        this.securityContextRepository = securityContextRepository;
+        this.authenticationManagementService = authenticationManagementService;
     }
 
-    /**
-     * Configure http security for OAuth 2.0 authentication, URL exceptions, CSRF tokens, etc.
-     * @param http HTTP security.
-     * @throws Exception Thrown when configuration fails.
-     */
-    @Override
-    protected void configure(HttpSecurity http) throws Exception {
+    @Bean
+    public SecurityFilterChain securityFilterChain(HttpSecurity http)
+            throws Exception {
+        final OAuth2AuthorizationServerConfigurer authorizationServerConfigurer = new OAuth2AuthorizationServerConfigurer();
+        authorizationServerConfigurer
+                .authorizationEndpoint(authorizationEndpoint ->
+                        authorizationEndpoint
+                                .authorizationResponseHandler((request, response, authentication) -> {
+                                    // Clear the security context just before the redirect
+                                    logger.info("Clearing security context after successful OAuth 2.1 authorization");
+                                    authenticationManagementService.clearContext();
+
+                                    // Redirect user to the original application.
+                                    logger.info("Redirecting user back to the original application");
+                                    redirectToOriginalApplication(request, response, authentication);
+                                })
+                );
         http
-                .csrf().ignoringAntMatchers("/api/auth/token/app/**", "/api/push/**", "/pa/**", OAUTH2_TOKEN_REVOKE_ENDPOINT)
+                // Apply OAuth 2.1 authorization server configuration
+                .apply(authorizationServerConfigurer)
                 .and()
-                    .antMatcher(OAUTH2_TOKEN_REVOKE_ENDPOINT).httpBasic().realmName(OAUTH2_CLIENT_REALM_NAME)
+                // Accept access tokens for user info endpoints in resource server, use token introspection
+                .oauth2ResourceServer((oauth2) -> oauth2
+                        .opaqueToken((opaque) -> opaque
+                                .introspectionUri(this.introspectionUri)
+                                .introspectionClientCredentials(this.clientId, this.clientSecret)
+                        )
+                )
+                // Configure securityContextRepository for session managemet, see: https://docs.spring.io/spring-security/reference/migration/servlet/session-management.html
+                .securityContext((securityContext) -> securityContext
+                        .securityContextRepository(securityContextRepository)
+                )
+                .csrf()
```

Comment:
It is possible to get rid of and() but we may postpone to a different pull request https://spring.io/blog/2019/11/21/spring-security-lambda-dsl

Reply:
Sure, why not.